modules/sugar/src/main/ts/ephox/sugar/api/node/ShadowDom.ts

```diff
@@ -70,3 +70,31 @@ export const getShadowRoot = (e: Element<DomNode>): Option<Element<ShadowRoot>>
  */
 export const getShadowHost = (e: Element<ShadowRoot>): Element<DomElement> =>
   Element.fromDom(e.dom().host);
+
+/**
+ * When Events bubble up through a ShadowRoot, the browser changes the target to be the shadow host.
+ * This function gets the "original" event target if possible.
+ * This only works if the shadow tree is open - if the shadow tree is closed, event.target is returned.
+ * See: https://developers.google.com/web/fundamentals/web-components/shadowdom#events
+ */
+export const getOriginalEventTarget = (event: Event): Option<EventTarget> => {
+  if (isSupported()) {
+    // When target element is inside Shadow DOM we need to take first element from composedPath
+    // otherwise we'll get Shadow Root parent, not actual target element.
+    // TODO: update @ephox/dom-globals to include Event.composedPath
+    const eventAny = event as any;
+    if (eventAny.composed && eventAny.composedPath) {
+      const composedPath = eventAny.composedPath();
+      if (composedPath && composedPath.length > 0) {
+        return Option.from(composedPath[0]);
+      }
+    }
+  }
+  return Option.from(event.target);
+};
+
+export const isOpen = (sr: Element<ShadowRoot>): boolean =>
+  (sr.dom() as any).mode === 'open';
```

Comment:
Worth noting if we upgrade the dom-globals version, you won't need the any here as well.

Reply:
Added TODO.